telegram/request/_requestdata.py

```diff
@@ -60,9 +60,9 @@ def parameters(self) -> Dict[str, Union[str, int, List[Any], Dict[Any, Any]]]:
             Dict[:obj:`str`, Union[:obj:`str`, :obj:`int`, List[any], Dict[any, any]]]
         """
         return {
-            param.name: param.value  # type: ignore[misc]
+            param.name: param.value
             for param in self._parameters
-            if param.value is not None
+            if param.value is not None and isinstance(param.value, (str, int, list, dict))
```

Comment:
Sorry, but this doesn't make it much better. This way, any other object would just silently be skipped. From my pov there are two scenarios here:

1. There actually are cases where param.value is none of string, int, list or dict. In that case we should make it obvious to the user that something went wrong. At the very least a warning logging. 
2. (I strongly suspect that this is the case here) Param.value will always be one of the documented types and the type hints as "object" is imprecise. In this case it's not a good solution to add _runtime_ checks to improve type checking. Instead, the typing in param.value would need be made more precise. This would also affect all places that build up in it.

I'm open to addressing 1/2 whichever is applicable or leaving the type-ignore comment in place for now.